src/syntax/syntax.rkt

```diff
@@ -381,10 +387,55 @@
                            (define-values (_ exs) (real-apply compiler pt))
                            (if exs (first exs) fail))
                          name)]))
+
+  ; make hash table
+  (define rules (make-hasheq))
+  (define count 0)
+  (define commutes? #f)
+  (when identities
+    (for ([ident (in-list identities)])
```

Comment:
Could also iterate over `(in-naturals)` instead of using mutable count. Then `(format ~a-exact-~a name i)` simply says that it's the "ith" rule.